ThreeBotPackages/tfgrid_solutions/tfgrid_solutions/chatflows/your_flist.py

```diff
@@ -99,9 +105,30 @@ def container_farm(self):
         else:
             query["hru"] = storage_units
         farms = j.sal.reservation_chatflow.farm_names_get(1, self, currency=self.currency, **query)
-        self.node = j.sal.reservation_chatflow.nodes_get(
-            1, farm_names=farms, currency=self.currency, **query
-        )[0]
+        self.node = j.sal.reservation_chatflow.nodes_get(1, farm_names=farms, currency=self.currency, **query)[0]
+
+    @j.baseclasses.chatflow_step(title="Attach Volume")
+    def container_volume(self):
+        volume_attach = self.drop_down_choice(
+            "Would you like to attach an extra volume to the container", ["YES", "NO"], required=True, default="NO"
+        )
+        self.container_volume_attach = volume_attach == "YES" or False
+
+    @j.baseclasses.chatflow_step(title="Volume storage type")
+    def container_volume_storage_type(self):
+        if self.container_volume_attach:
+            self.user_form_data["Volume Disk type"] = self.drop_down_choice(
+                "Please choose the type of disk for the volume", ["SSD", "HDD"], required=True, default="SSD"
+            )
+
+    @j.baseclasses.chatflow_step(title="Volume Size")
+    def container_volume_size(self):
```

Comment:
please ask this in the same 'form' as the type

Reply:
Fix: https://github.com/threefoldtech/jumpscaleX_threebot/pull/746/commits/fe4df0f9af97f4d798e11a3965e87ed4fcb50752#diff-e27990502904cf31b839b628ec93acc6R117